course-v2/assets/js/video_download_popup.js

```diff
@@ -1,16 +1,10 @@
 export const initVideoDownloadPopup = () => {
-  const downloadIcon = document.querySelector(".video-download-icons")
+  const downloadIcons = document.querySelector(".video-download-icons")
   const popup = document.querySelector(".video-tab-download-popup")
-  if (downloadIcon) {
-    downloadIcon.addEventListener("click", event => {
+  if (downloadIcons) {
+    downloadIcons.addEventListener("click", event => {
       event.stopPropagation()
       if (popup) popup.classList.toggle("hidden")
     })
   }
-
-  document.addEventListener("click", () => {
-    if (popup) {
-      popup.classList.add("hidden")
-    }
-  })
```

Comment:
Was this a temporary change? I think it should come back.

Reply:
This wasn't a temporary change. After changing the `divs` to `buttons`/ `links` (with the old current refactoring of the code), without this change the download button was not working (the popup would not open)

This has been a while so I'm not sure why exactly was this causing that.
However,  since then to this point, I have refactored a bunch of stuff, and I just checked by re-adding the event listener and it seems to be working.

So I will put this back after testing it.

Reply:
Do double check it. This is the chunk of code that is responsible for "Close popup when click outside of popup". So that's not working on this branch right now.